editing.py

```diff
@@ -175,6 +175,45 @@ def add_artist(self, artist, edit_note, auto=False):
         self.b.submit()
         return self._extract_mbid("artist")
 
+    def add_area(self, area, edit_note, auto=False):
+        # Unlike other methods, this one directly requests page using
+        # mechanize.Requests to overcome mechanize's lack of javascript support
+
+        data_template = {
+            "edit-area.name": area["name"],
+            "edit-area.comment": area["disambiguation"],
+            "edit-area.type_id": area["type"],
+            "edit-area.edit_note": edit_note,
+            "edit-area.iso_3166_1.0": area["iso_3166_1"],
+            "edit-area.iso_3166_2.0": area["iso_3166_2"],
+            "edit-area.iso_3166_3.0": area["iso_3166_3"],
+            "edit-area.url.0.text": area["external_links"]["text"],
+            "edit-area.url.0.link_type_id": area["external_links"]["type"],
+        }
+
+        def create_payload(template):
+            # remove null values from template
+            template = {k: v for k, v in template.items() if v is not None}
```

Comment:
I'd rather we throw an error if certain basic required fields like `name` are `None`. (You'll have to check which other fields are required.)